packages/foam-vscode/src/extension.ts

```diff
@@ -120,6 +127,22 @@ async function updateReferenceList() {
   }
 }
 
+async function openDailyNote() {
+  const date = dateFormat(new Date(), "isoDate");
+  
+  const fileName = `${date}.md`;
+  const dailyNotePath = join(workspace.workspaceFolders[0].uri.fsPath, fileName);
+  
+  if (!existsSync(dailyNotePath)) {
+    writeFileSync(dailyNotePath, `# ${date}\r\n`);
+  }
+
+  var openPath = Uri.parse(dailyNotePath);
+  workspace.openTextDocument(openPath).then(doc => {
+    window.showTextDocument(doc);
+  });
```

Comment:
nit: I don't know if you grabbed this from [Stack Overflow](https://stackoverflow.com/a/45093852), but if we leave as is, I think we should leave a comment and credit the original author. Thoughts?

Otherwise, nit/suggestion?:

```suggestion
  const openPath = Uri.parse(dailyNotePath);
  
  const doc = await workspace.openTextDocument(openPath);
  window.showTextDocument(doc);
```

Reply:
It would also be nice to add a comment explaining _why_ we have to `Uri.parse(dailyNotePath)` insated of passing `dailyNotePath` to `workspace.openTextDocument` (I feel like future contributors will thank you)

Reply:
You're right about giving credit to the author if we keep it as is. I'll also prefer the `async` way. For the `Uri.parse()` I have to admit that I don't know if it's needed but I'll take a look at it and explain further if we need it.

Thanks for your review, it's appreciated!

Reply:
@jsjoeio I pushed a fixup commit. I changed it to the `async` way but I won't add a comment concerning the `Uri.parse()` since it's the type that `workspace.openTextDocument` expects so it is self documenting.

Reply:
Awesome! Thanks for doing that. 

> I won't add a comment concerning the Uri.parse() since it's the type that workspace.openTextDocument expects so it is self documenting

Ah, right, we can rely on the type definition. Good call!